### DISCUSSION TITLE
k8s log show  error" tcp is not a valid port number"

### DISCUSSION THREAD

**Discussion by scany1211** [apache/superset#10806] [General]
hi experts,
    I download the superset source code and try to build the docker image with command "docker build -t xxx:xx .", then upload the image to docker registry and create the K8S pod. By the way,  I create the K8S pod with offical superset images, it has no problem. Only having  the problem when I use the image built by  myself.
   Do I need to modify anything in the source code ?

Thanks

**Comment by madelfio** [comment]
I had the same issues.  I realized I had a Kubernetes/OpenShift service named "superset", which appears to cause a `SUPERSET_PORT` environment variable to be auto-injected into the container, overriding the value specified in the superset Dockerfile.  Once I deleted/renamed that service, I got past the `tcp is not a valid port number` error.

**Reply by kwims** [reply]
@madelfio , thanks you just saved me a lot of time :)

**Reply by metasim** [reply]
ditto!!!

**Reply by johnnycube** [reply]
@madelfio thank you very much - solved a longstanding issue I had

**Reply by abhikashyap** [reply]
you saved me 
i was struggling from around 10 hours 
thank you so much

**Comment by sunkyzz** [comment]
I also meet the same issue, the reason is service `superset` will override `SUPERSET_PORT` env var like `tcp://123.233.1.1:8088` instead of `8088`. Rename service or edit `docker-entrypoint.sh` in docker image will solve this issue ;)

**Comment by arthursimas1** [comment]
you can also set an environment variable `SUPERSET_PORT` to the pod that will override the value set by the service

**Comment by mistercrunch** [comment]
anything we should do in this repo to prevent people running into this?
